woocommerce/woocommerce-android
# Product list selection

Closes: #7929
Closes: #7930

### Why
To be able able to bulk update products, first, the merchant needs to select the products he wants to update.

### Description
This task is about enabling a selectable state of the products list view. It allows product selection by using the `SelectionTracker`, part of `androidx.recyclerview.selection`, an addon library providing support for item selection.

### Testing instructions
TC1 
1. Open the products list screen
2. Tap on the multi-select menu item
3. Check that the view enters on selection mode
4. Tap on the products and check that the selection/deselection works as expected
5. Check that the selection count works as expected

TC2
1. Open the products list screen
2. Long press a product and check that the view enters on selection mode
3. Complete Steps 4 and 5 of TC1

TC3
1. Open the products list screen
2. Tap on the multi-select menu item
3. Tap on the more menu (3 dots)
4. Tap on Select All
5. Check that all loaded products are selected

TC4
1. Open the products list screen
2. Tap on Search
3. Type a valid search query
4. Long press a product to enter selection mode
5. Tap on <- to exit the selection mode
6. Tap on <- to exit the search
7. Check that the app doesn't crash.
8. Check [this comment](https://github.com/woocommerce/woocommerce-android/pull/7971#issuecomment-1341503152)

### Images/gif
https://user-images.githubusercontent.com/18119390/205197204-0fde6a4a-3c57-4410-ad7b-639bfb46b668.mp4
